buildscripts/cmake/SetupBuildEnvironment.cmake

```diff
@@ -1,87 +1,95 @@
+
+if(MUE_COMPILE_USE_SHARED_LIBS_IN_DEBUG AND BUILD_IS_DEBUG)
+    if(CC_IS_GCC OR CC_IS_MINGW)
         set(BUILD_SHARED_LIBS ON)
     endif()
+endif()
 
-    if (MUSE_COMPILE_ASAN)
+# Address Sanitizer
+if(MUSE_COMPILE_ASAN)
+    if(CC_IS_CLANG OR CC_IS_GCC OR CC_IS_MINGW)
         add_compile_options("-fsanitize=address")
         add_compile_options("-fno-omit-frame-pointer")
-        link_libraries("-fsanitize=address")
-    endif()
-
-elseif(CC_IS_MSVC)
-    message(STATUS "Using Compiler MSVC ${CMAKE_CXX_COMPILER_VERSION}")
-
-    set(CMAKE_CXX_FLAGS                 "/MP /EHsc /utf-8")
-    set(CMAKE_C_FLAGS                   "/MP /utf-8")
-    set(CMAKE_CXX_FLAGS_DEBUG           "/MT /Zi /Ob0 /Od /RTC1")
-    set(CMAKE_CXX_FLAGS_RELEASE         "/MT /O2 /Ob2")
-    set(CMAKE_CXX_FLAGS_RELWITHDEBINFO  "/MT /Zi /O2 /Ob1")
-    set(CMAKE_C_FLAGS_DEBUG             "/MT /Zi /Ob0 /Od /RTC1")
-    set(CMAKE_C_FLAGS_RELEASE           "/MT /O2 /Ob2")
-    set(CMAKE_C_FLAGS_RELWITHDEBINFO    "/MT /Zi /O2 /Ob1")
-
-    add_compile_definitions(WIN32)
-    add_compile_definitions(_WINDOWS)
-    add_compile_definitions(_UNICODE)
-    add_compile_definitions(UNICODE)
-    add_compile_definitions(_USE_MATH_DEFINES)
-    add_compile_definitions(NOMINMAX)
-
-elseif(CC_IS_MINGW)
-    message(STATUS "Using Compiler MINGW ${CMAKE_CXX_COMPILER_VERSION}")
-
-    set(CMAKE_CXX_FLAGS_DEBUG   "-g")
-    set(CMAKE_CXX_FLAGS_RELEASE "-O2")
-
-    if (MUE_COMPILE_USE_SHARED_LIBS_IN_DEBUG AND BUILD_IS_DEBUG)
-        set(BUILD_SHARED_LIBS ON)
-    endif()
-
-    # -mno-ms-bitfields see #22048
-    set(CMAKE_CXX_FLAGS         "${CMAKE_CXX_FLAGS} -mno-ms-bitfields")
-    if (NOT MUSE_COMPILE_BUILD_64)
-        set(CMAKE_EXE_LINKER_FLAGS "-Wl,--large-address-aware")
+        add_link_options("-fsanitize=address")
+    elseif(CC_IS_MSVC)
+        add_compile_options("/fsanitize=address")
+        add_link_options("/fsanitize=address")
     endif()
+endif()
 
-    add_compile_definitions(_UNICODE)
-    add_compile_definitions(UNICODE)
+# Mac-specific
+if(OS_IS_MAC)
+    set(MACOSX_DEPLOYMENT_TARGET 10.14)
+    set(CMAKE_OSX_DEPLOYMENT_TARGET 10.14)
+endif(OS_IS_MAC)
 
-elseif(CC_IS_CLANG)
-    message(STATUS "Using Compiler CLANG ${CMAKE_CXX_COMPILER_VERSION}")
+# MSVC-specific
+if(CC_IS_MSVC)
+    add_compile_options("/EHsc")
+    add_compile_options("/utf-8")
 
-    set(CMAKE_CXX_FLAGS_DEBUG   "-g")
-    set(CMAKE_CXX_FLAGS_RELEASE "-O2")
+    add_compile_definitions(WIN32 _WINDOWS)
+    add_compile_definitions(_UNICODE UNICODE)
+    add_compile_definitions(_USE_MATH_DEFINES)
+    add_compile_definitions(NOMINMAX)
+endif()
 
-    if (MUSE_COMPILE_ASAN)
-        add_compile_options("-fsanitize=address")
-        add_compile_options("-fno-omit-frame-pointer")
-        link_libraries("-fsanitize=address")
-    endif()
+# MinGW-specific
+if(CC_IS_MINGW)
+    # https://musescore.org/node/22048
+    add_compile_options(-mno-ms-bitfields)
```

Comment:
As you can tell from the 5-digit .org node number, this is really old; but it looked like it might still be relevant

Reply:
Well, that is/was for MINGW only, and that doesn't work anymore (and since long, since the new sound engine stuff)

Reply:
So as far as I can tell the MINGW stuff can go entirely.
Unless someone explains how to build using MINGW...

Reply:
I chose not (yet) to delete MinGW support completely. Perhaps we can still fix it someday...